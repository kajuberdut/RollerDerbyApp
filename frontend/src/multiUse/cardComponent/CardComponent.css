.CardComponent {
    background-color: #d1d2d4;
    /* margin-top: 150px; */
    margin-top: 40px;
    height: 250px;
    width: 43%;
    margin-left: 12%;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
}

.CardComponent-Title h3:hover,
.CardComponent-Title h3[data-hover] {
  color: black; 
}