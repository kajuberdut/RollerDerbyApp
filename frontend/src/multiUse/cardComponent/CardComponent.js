import React, { useState, useContext, useEffect } from "react";
import { NavLink } from "react-router-dom";
import FastApi from "../../Api";
import "./CardComponent.css";
import Loading from "../loading/Loading"
import { Card, CardBody, CardTitle, CardText } from "reactstrap";

  /**  
  * Card component for bouts and mixers. 
  */

  function CardComponent({mixer, bout}) {

  /** Set address and is loading in state */ 
    const [address, setAddress ] = useState([]); 
    const [isLoading, setIsLoading] = useState(true);

    let info = bout !== undefined ? bout : mixer;
    console.log("info:", info)
    let type = bout !== undefined ? "bouts" : "mixers";
    console.log("*****************************************")
    console.log("type", type)

    let key = info.eventId
    
  /** API get request for a specific address */ 

    async function getAddress() {
   
      let address = await FastApi.getAddress(info.addressId)
      console.log("address:", address)
      setAddress(address)
      setIsLoading(false)
    }
    
  /** If bout or mixer changes call get address again */ 

    useEffect(() => {
      getAddress();
    }, [bout, mixer]);


      /** Display loading if API call is has not returned */

    if (isLoading) {
      return (
          <Loading />
      )
    }   
  
     /** Render the card component */
      
      return (
          <section key={"CardComponent-" + key}>
            <Card className="CardComponent"> 
              <CardBody>
                <NavLink to={`/events/${type}/${key}`} className="CardComponent-Link" style={{color: '#555555', textDecoration: 'none'}}>
                  <CardTitle className="text-center CardComponent-Title">
                    <h3>{info.theme}</h3>
                  </CardTitle>
                  <CardTitle> 
                  <h5>{address.city}, {address.state}</h5>
                  </CardTitle>
                  <CardText className="CardComponent-Text">
                  {info.date}
                  </CardText>
                  <CardText className="CardComponent-Text">
                    {info.description}
                  </CardText>
                </NavLink>
              </CardBody>
            </Card>
         </section>
        );
  }
  
  export default CardComponent
