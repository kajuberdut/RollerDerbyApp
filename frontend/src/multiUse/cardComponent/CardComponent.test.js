import { render, screen, waitFor } from '@testing-library/react';
import CardComponent from './CardComponent.js';
import { BrowserRouter } from "react-router-dom";
import FastApi from '../../Api'; 

// * Passing as of 2/14/24
// npm test CardComponent.test.js
// must be in frontend directory 

// Mock FastApi methods
jest.mock('../../Api', () => ({
    getAddress: jest.fn()
  }));

// Mock useParams
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), 
    useParams: jest.fn(),
  }));

  let bout = {
    
        eventId: 1,
        type: "bout",
        date: "2024-01-25",
        addressId: 1,
        time: "19:10",
        timeZone: "Mountain Time (MT): America/Denver (Denver, Phoenix, Salt Lake City)",
        theme: "Theme Test 1",
        description: "Come have a blast and skate with us. It will be the ultimate skate party! ",
        level: "All Levels",
        coEd: false,
        ruleset: "WFTDA",
        floorType: "Slick",
        jerseyColors: "Black and White",
        groupId: 2,
        chatId: 2,
        opposingTeam: "Boulder",
        team: "Cheyenne Capidolls"
  }
  
describe("checks rendering of card component", () => {  

    test('renders card component form page', () => {

        render(
            <BrowserRouter>
                    <CardComponent bout={bout} />
            </BrowserRouter>);
        });


    test('matches snapshot', function() {
        const { asFragment } = render(
            <BrowserRouter>
                    <CardComponent bout={bout} />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot(); 
    })

    screen.debug()

});

describe("asserts card component text", () => {  

    const mockAddress = {
        streetAddress: '123 Main St',
        city: 'Anytown',
        state: 'State',
        zipCode: '12345',
      };

      beforeEach(() => {
        FastApi.getAddress.mockResolvedValue(mockAddress);
      });
    
      afterEach(() => {
        jest.clearAllMocks();
      });
    

    test('renders card theme on page', async () => {

        render(
            <BrowserRouter>
                    <CardComponent bout={bout} mixer={{}} />
            </BrowserRouter>
        )

        await waitFor(() => {
            expect(screen.getByText('Theme Test 1')).toBeInTheDocument();
            expect(screen.getByText('2024-01-25')).toBeInTheDocument();
            
        })

    });

    test('renders address on page', async () => {

        render(
            <BrowserRouter>
                    <CardComponent bout={bout} mixer={{}} />
            </BrowserRouter>
        )

        await waitFor(() => {
            expect(screen.getByText('Anytown, State')).toBeInTheDocument();
            expect(screen.getByText('Come have a blast and skate with us. It will be the ultimate skate party!')).toBeInTheDocument();
        })
    });

});