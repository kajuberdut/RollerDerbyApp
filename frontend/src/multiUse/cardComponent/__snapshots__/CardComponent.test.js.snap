// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of card component matches snapshot 1`] = `<DocumentFragment />`;
